.github/scripts/cmd/cmd.py

```diff
@@ -0,0 +1,204 @@
+        elif args.runtime and not args.pallet:
+            print(f"{args.runtime} runtime does not have any pallets")
+        elif args.runtime and args.pallet:
+            print(f"No pallets [{args.pallet}] found in {args.runtime}")
+        else:
+            print('No runtimes found')
+        sys.exit(0)
+
+    header_path = os.path.abspath('./.github/scripts/cmd/file_header.txt')
+
+    for runtime in runtime_pallets_map:
+        for pallet in runtime_pallets_map[runtime]:
+            config = runtimesMatrix[runtime]
+            print(f'-- config: {config}')
+            if runtime == 'dev':
+                # to support sub-modules (https://github.com/paritytech/command-bot/issues/275)
+                search_manifest_path = f"cargo metadata --format-version 1 --no-deps | jq -r '.packages[] | select(.name == \"{pallet.replace('_', '-')}\") | .manifest_path'"
+                print(f'-- running: {search_manifest_path}')
+                manifest_path = os.popen(search_manifest_path).read()
+                package_dir = os.path.dirname(manifest_path)
+                print(f'-- package_dir: {package_dir}')
+                print(f'-- manifest_path: {manifest_path}')
+                output_path = os.path.join(package_dir, "src", "weights.rs")
+            else:
+                default_path = f"./{config['path']}/src/weights"
+                xcm_path = f"./{config['path']}/src/weights/xcm"
+                output_path = default_path if not pallet.startswith("pallet_xcm_benchmarks") else xcm_path
+            print(f'-- benchmarking {pallet} in {runtime} into {output_path}')
+            cmd = f"frame-omni-bencher v1 benchmark pallet --extrinsic=* --runtime=target/{profile}/wbuild/{config['package']}/{config['package'].replace('-', '_')}.wasm --pallet={pallet} --header={header_path} --output={output_path} --wasm-execution=compiled --steps=50 --repeat=20 --heap-pages=4096"
+            print(f'-- Running: {cmd}')
+            status = os.system(cmd)
+            if status != 0 and not args.continue_on_fail:
+                print(f'Failed to benchmark {pallet} in {runtime}')
+                sys.exit(1)
+
+            # Otherwise collect failed benchmarks and print them at the end
+            # push failed pallets to failed_benchmarks
+            if status != 0:
+                failed_benchmarks[f'{runtime}'] = failed_benchmarks.get(f'{runtime}', []) + [pallet]
+            else:
+                successful_benchmarks[f'{runtime}'] = successful_benchmarks.get(f'{runtime}', []) + [pallet]
+
+    if failed_benchmarks:
+        print('❌ Failed benchmarks of runtimes/pallets:')
+        for runtime, pallets in failed_benchmarks.items():
+            print(f'-- {runtime}: {pallets}')
+
+    if successful_benchmarks:
+        print('✅ Successful benchmarks of runtimes/pallets:')
+        for runtime, pallets in successful_benchmarks.items():
+            print(f'-- {runtime}: {pallets}')
+
+    tempdir.cleanup()
```

Comment:
Wont be cleaned up in the error case. Maybe wrapping it in a function would help.

Reply:
actually we don't need tempdir there, i will remove it :)